### DISCUSSION TITLE
Using the wrong username breaks things

### DISCUSSION THREAD

**Discussion by tareefdev** [magit/forge#606] [Support] [answered]
Two weeks ago Forge does not pull any new data from Gitlab. I run the command, it says 'Pulling user/project..' and nothing more. Is there a flag like --verbose for Forge?

**Comment by tarsius** [comment]
I seem to be having the same issue. I'll look into it once I have landed a huge feature branch I've been working on for a while now.

**Reply by tareefdev** [reply]
I don't want to be annoying, but I'm wondering when we can expect this issue to be resolved or how I can help. I am not a Lisp programmer, but I can try my best if you guide me on the issue.

**Reply by tarsius** [reply]
Ah, for me that worked again the next time I tried, so I assumed it was some temporary issue on their end and forgot about this. I'm afraid I won't have any time to investigate issues such as this until the new year. I don't have any quick advice what you could try at this time, except maybe try with another repository and see if you can narrow it down to, e.g., large repositories.

**Reply by tareefdev** [reply]
💐 :)

**Comment by tareefdev** [comment, marked as answer]
It worked again. I discovered that my Gitlab username in `~/.gitconfig` was wrong; while it was correct in `~/.authinfo,` this mismatch somehow introduced, and maybe it caused the above-mentioned issue.

**Comment by Niols** [comment]
This question pops up when trying to debug Forge, and it does not answer on that point. Would you mind either renaming the discussion, or (better) link to the appropriate piece of documentation, or maybe other discussion, that does explain how to debug things?

**Reply by tarsius** [reply]
> This question pops up when trying to debug Forge, and it does not answer on that point.

Yes, sometimes search engines link to things that are not actually relevant to what the user was looking for.

Debugging Emacs often involves `M-x toggle-debug-on-error`, but other times there is an issue and yet that does not result in an error, in which case you will have to use different strategies. What strategies are appropriate depends on the issue. Generally speaking investigating Forge issues is not different from investigating issues in any Emacs package, or even software in general.

**Reply by Niols** [reply]
> Yes, sometimes search engines link to things that are not actually relevant to what the user was looking for.

In this particular case, I searched how to debug Forge and stumbled on a discussion named “how to debug forge” on the Forge GitHub repository; that sounds very relevant.

Thank you for your insights on debugging forge!

**Reply by tarsius** [reply]
I've now decided to edit the title.

**Reply by Niols** [reply]
Thanks :-)
